src/Avalonia.Base/Data/Core/SettableNode.cs

```diff
@@ -19,11 +19,25 @@ private bool ShouldNotSet(object value)
             {
                 return false;
             }
+
+            if (LastValue == null)
```

Comment:
This is the only actual change in the logic, right?

Reply:
Yes, there is also a small changes in `Gestures.cs` that fixes potential data race there. But generally everything else is just changing from `WeakReference` to `WeakReference<T>` which goes quite deep so this PR has a lot of changes.